orthoinference/src/main/java/org/reactome/orthoinference/InferEWAS.java

```diff
@@ -0,0 +1,364 @@
+	public static ArrayList<GKInstance> createReferenceDNASequence(String homologueId) throws InvalidAttributeException, InvalidAttributeValueException, Exception
+	{
+		ArrayList<GKInstance> referenceDNAInstances = new ArrayList<GKInstance>();
+		ArrayList<String> ensgs = ensgMappings.get(homologueId);
+		for (Object ensg : ensgs)
+		{
+			SchemaClass referenceDNAClass = dba.getSchema().getClassByName(ReactomeJavaConstants.ReferenceDNASequence);
+			GKInstance referenceDNAInst = new GKInstance(referenceDNAClass);
+			referenceDNAInst.setDbAdaptor(dba);
+			referenceDNAInst.addAttributeValue(ReactomeJavaConstants.created, instanceEdit);
+			referenceDNAInst.addAttributeValue(ReactomeJavaConstants.identifier, ensg);
+			referenceDNAInst.addAttributeValue(ReactomeJavaConstants.referenceDatabase, ensgDbInst);
+			referenceDNAInst.addAttributeValue(ReactomeJavaConstants.species, speciesInst);
+			referenceDNAInst.setAttributeValue(ReactomeJavaConstants._displayName, "ENSEMBL:" + ensg);
+			referenceDNAInst = GenerateInstance.checkForIdenticalInstances(referenceDNAInst);
+			referenceDNAInstances.add(referenceDNAInst);
+			if (refDb)
+			{
+				GKInstance alternateRefDNAInst = new GKInstance(referenceDNAClass);
+				alternateRefDNAInst.setDbAdaptor(dba);
+				String altDbIdentifier = (String) ensg;
+				if (altRefDBId != null) {
+					altDbIdentifier = altDbIdentifier.replaceAll(altRefDBId, "");
+				}
+				alternateRefDNAInst.addAttributeValue(ReactomeJavaConstants.created, instanceEdit);
+				alternateRefDNAInst.addAttributeValue(ReactomeJavaConstants.identifier, altDbIdentifier);
+				alternateRefDNAInst.addAttributeValue(ReactomeJavaConstants.referenceDatabase, alternateDbInst);
+				alternateRefDNAInst.addAttributeValue(ReactomeJavaConstants.species, speciesInst);
+				alternateRefDNAInst.setAttributeValue(ReactomeJavaConstants._displayName, alternateDbInst.getAttributeValue(ReactomeJavaConstants.name) + ":" + ensg);
+				alternateRefDNAInst = GenerateInstance.checkForIdenticalInstances(alternateRefDNAInst);
+				referenceDNAInstances.add(alternateRefDNAInst);
+			}
+		}
+		return referenceDNAInstances;
+	}
+	
+    // These are setup functions called at the beginning of the 'inferEvent' script 
+	
+	public static void setAdaptor(MySQLAdaptor dbAdaptor)
+	{
+		InferEWAS.dba = dbAdaptor;
+	}
+	// Sets the HashMap of species-specific homologue-identifier mappings
+	public static void setHomologueMappingFile(HashMap<String, String[]> homologueMappingsCopy) throws IOException
+	{
+		homologueMappings = homologueMappingsCopy;
+	}
+	
+	// Read the species-specific ENSG gene-protein mappings, and create a Hashmap with the contents
+	public static void readENSGMappingFile(String toSpecies) throws IOException
+	{
+		String mappingFileName = toSpecies + "_gene_protein_mapping.txt";
+		String mappingFilePath = "src/main/resources/orthopairs/" + mappingFileName;
```

Comment:
Will this file _always_ be included in src/main/resources? Or would it be better to make this configurable? I don't know enough about this to know if it's OK to hard-code this path like this, or if it should be more flexible/configurable...

Reply:
So it will be configurable for release, since the Orthopairs files will be under /orthopairs/{releaseNumber} -- just temporarily hard-coded in my workspace

Reply:
Ok... Just remember that the final slice is supposed to happen on Wednesday, and OrthoInference will happen shortly after (Thursday or Friday, probably). I hope you'll have enough time to code and test the changes necessary to get it working on the reactomerelease!